### PR TITLE
tapchannel: add 30 second timeout for funding ack

### PR DESCRIPTION
**Pull request by Roasbeef**
This ensures that the CLI won't hang indefinitely if the other party fails to respond for we/ reason (disconnected TCP connection, etc).

### CODE REVIEW

File: tapchannel/aux_funding_controller.go [hunk 1]
```diff
@@ -1107,6 +1112,11 @@ func (f *FundingController) completeChannelFunding(ctx context.Context,
 	log.Debugf("Waiting for channel finalized signal...")
 	select {
 	case <-fundingState.fundingFinalizedSignal:
+
+	case <-time.After(ackTimeout):
+		return fmt.Errorf("didn't receive funding ack after %v "+
```
[the remark this answers is not in the document]
Comment (Roasbeef) [reply tapchannel/aux_funding_controller.go]: `gopls crashed in vim for some reason, so didn't pick this up, fixed!

File: tapchannel/aux_funding_controller.go [hunk 2]
```diff
@@ -1283,6 +1293,14 @@ func (f *FundingController) chanFunder() {
 						return
 					}
 
+				case <-time.After(ackTimeout):
+					err := fmt.Errorf("didn't receive "+
+						"funding ack after %v seconds",
```
[the remark this answers is not in the document]
Comment (jharveyb) [reply tapchannel/aux_funding_controller.go:1298]: Not resolved?

**Review by jharveyb — APPROVED** [review]
LGTM 👍🏽

File: tapchannel/aux_funding_controller.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (guggero) [reply tapchannel/aux_funding_controller.go]: Pushed a fix.

**Review by guggero — APPROVED** [review]
LGTM :tada:

**Pull request merged**